gear-programs/vft-manager/app/src/services/bridge_builtin_operations.rs

```diff
@@ -0,0 +1,55 @@
+use super::{msg_tracker_mut, utils, Config, Error, MessageStatus};
+use gstd::MessageId;
+use sails_rs::prelude::*;
+
+pub async fn send_message_to_bridge_builtin(
+    gear_bridge_builtin: ActorId,
+    receiver_contract_address: H160,
+    payload: Payload,
+    config: &Config,
+    msg_id: MessageId,
+) -> Result<U256, Error> {
+    msg_tracker_mut().update_message_status(msg_id, MessageStatus::SendingMessageToBridgeBuiltin);
+
+    let payload_bytes = payload.pack();
+
+    let bytes = gbuiltin_eth_bridge::Request::SendEthMessage {
+        destination: receiver_contract_address,
+        payload: payload_bytes,
+    }
+    .encode();
+
+    utils::set_critical_hook(msg_id);
+    utils::send_message_with_gas_for_reply(
+        gear_bridge_builtin,
+        bytes,
+        config.gas_to_send_request_to_builtin,
+        config.gas_for_reply_deposit,
+        config.reply_timeout,
+        msg_id,
+    )
+    .await?;
+    msg_tracker_mut().check_bridge_reply(&msg_id)
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo)]
+pub struct Payload {
+    pub receiver: H160,
+    pub token_id: H160,
+    pub amount: U256,
+}
+
+impl Payload {
+    pub fn pack(self) -> Vec<u8> {
+        let mut packed = Vec::with_capacity(20 + 20 + 32); // H160 is 20 bytes, U256 is 32 bytes
```

Comment:
if i'm not mistaken there are no comments aside the code. Usually it places above(before) the next expression.